src/DataLayouts/DataLayouts.jl

```diff
@@ -491,6 +491,10 @@ end
 @inline function Base.getindex(data::IFH{S}, i, _, _, _, h) where {S}
     @inbounds get_struct(parent(data), S, Val(2), CartesianIndex(i, 1, h))
 end
+@inline function Base.getindex(data::IFH{S}, I::CartesianIndex{5}) where {S}
```

Comment:
I'm really happy about these additions (which, IIUC, were added due to removing `getidx(column(field, hidx...), loc, idx)` below). I think this is generally a better memory access pattern, it reminds me of performance benefits of `a[i,j]` over `a[i][j]`.